src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/EvaluateConcordanceOnNonGenotypedSVVCFs.java

```diff
@@ -0,0 +1,497 @@
+                if (svLen < StructuralVariationDiscoveryArgumentCollection.STRUCTURAL_VARIANT_SIZE_LOWER_BOUND) {
+                    evalSkippedTesting.add( new VariantContextBuilder(vc).attribute(VARIANT_SKIPPED_EVAL_REASON_KEY, "tiny_ins").make() );
+                    return false;
+                }
+            }
+
+            return true;
+        };
+    }
+
+    /**
+     * A 2-step evaluation mechanism:
+     * <ul>
+     *     <li> first test if {@code eval} (padded by {@link #getPaddedSvInterval(VariantContext)}) overlaps with any record in the truth tree </li>
+     *     <li> further test concordance if there's such overlap </li>
+     * </ul>
+     */
+    @Override
+    protected void apply(final VariantContext eval, final ReadsContext readsContext, final ReferenceContext refContext) {
+        final SVInterval evalInterval = getPaddedSvInterval(eval);
+        if ( truthTree.hasOverlapper(evalInterval) ) {
+            final SupportingTruth supportingTruth = getConcordantTruthCalls(eval, evalInterval);
+            if (supportingTruth == null) {
+                classifiedVariants.add(new FalsePositive(eval, FalsePositive.ReasonForFalsePositive.OVERLAPPING_TRUTH_WITH_DIFF_TYPE_OR_LOW_SUPPORT));
+            } else {
+                classifiedVariants.add( eval.isFiltered() ? new FilteredFalseNegative(eval, supportingTruth.truthVar, supportingTruth.maxSupportScore)
+                                                          : new TruePositive(eval, supportingTruth.truthVar, supportingTruth.maxSupportScore));
+            }
+        } else {
+            if (eval.isFiltered()) {
+                classifiedVariants.add(new FilteredTrueNegative(eval));
+            } else {
+                classifiedVariants.add(new FalsePositive(eval, FalsePositive.ReasonForFalsePositive.NO_OVERLAPPING_TRUTH));
+            }
+        }
+    }
+
+    // TODO: 6/3/18 padding can be improved
+    @Override
+    protected SVInterval getPaddedSvInterval(final VariantContext eval) {
+
+        final int homologyLength = eval.getAttributeAsInt(GATKSVVCFConstants.HOMOLOGY_LENGTH, 0);
+
+        int insTypeRightPadding = 0;
+        final String svType = eval.getAttributeAsString(GATKSVVCFConstants.SVTYPE, "");
+        if (svType.equals(GATKSVVCFConstants.SYMB_ALT_ALLELE_INS) || svType.equals(GATKSVVCFConstants.SYMB_ALT_ALLELE_DUP))
+            insTypeRightPadding = eval.getAttributeAsInt(GATKSVVCFConstants.SVLEN, 0);
+
+        // we only pad to the right with homology because we expect variants to follow left-align convention (GATK-SV does this)
+        final int leftPadding = fixedPadding;
+        final int rightPadding = fixedPadding + homologyLength + insTypeRightPadding;
+        final int chromosomeLength = refSeqDict.getSequence(eval.getContig()).getSequenceLength();
+        return new SVInterval(refSeqDict.getSequenceIndex(eval.getContig()),
```

Comment:
SVInterval position first valid value is 1 not 0. The fact that its constructor does not check for it is kinda of buggy but perhaps a performance decision. So the first argument of ```max``` should be 1 and perhaps the second argument should not have the ```-1``` but I'm less sure of that one.

Reply:
corrected throughout